homeassistant/components/onewire/config_flow.py

```diff
@@ -0,0 +1,153 @@
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_TYPE
+from homeassistant.data_entry_flow import AbortFlow
+
+from .const import (  # pylint: disable=unused-import
+    CONF_MOUNT_DIR,
+    CONF_TYPE_OWFS,
+    CONF_TYPE_OWSERVER,
+    CONF_TYPE_SYSBUS,
+    DEFAULT_OWSERVER_HOST,
+    DEFAULT_OWSERVER_PORT,
+    DEFAULT_SYSBUS_MOUNT_DIR,
+    DOMAIN,
+)
+
+DATA_SCHEMA_USER = vol.Schema(
+    {vol.Required(CONF_TYPE): vol.In([CONF_TYPE_OWSERVER, CONF_TYPE_SYSBUS])}
+)
+DATA_SCHEMA_OWSERVER = vol.Schema(
+    {
+        vol.Required(CONF_HOST, default=DEFAULT_OWSERVER_HOST): str,
+        vol.Required(CONF_PORT, default=DEFAULT_OWSERVER_PORT): int,
+    }
+)
+DATA_SCHEMA_MOUNTDIR = vol.Schema(
+    {
+        vol.Required(CONF_MOUNT_DIR, default=DEFAULT_SYSBUS_MOUNT_DIR): str,
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class OneWireFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle 1-Wire config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize 1-Wire config flow."""
+        self.onewire_config = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle 1-Wire config flow start.
+
+        Let user manually input configuration.
+        """
+        errors = {}
+        if user_input is not None:
+            self.onewire_config.update(user_input)
+            if CONF_TYPE_OWSERVER == user_input[CONF_TYPE]:
+                return await self.async_step_owserver()
+            if CONF_TYPE_SYSBUS == user_input[CONF_TYPE]:
+                return await self.async_step_mount_dir()
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=DATA_SCHEMA_USER,
+            errors=errors,
+        )
+
+    def get_existing_owserver_entry(self, host: str, port: int):
+        """Get existing entry with matching host and port."""
+        for config_entry in self.hass.config_entries.async_entries(DOMAIN):
+            if (
+                config_entry.data[CONF_TYPE] == CONF_TYPE_OWSERVER
+                and config_entry.data[CONF_HOST] == host
+                and config_entry.data[CONF_PORT] == str(port)
+            ):
+                return config_entry
+        return None
+
+    async def async_step_owserver(self, user_input=None):
+        """Handle OWServer configuration."""
+        errors = {}
+        if user_input:
+            self.onewire_config.update(user_input)
+            owhost = user_input[CONF_HOST]
+            owport = user_input[CONF_PORT]
+            try:
+                existing_entry = self.get_existing_owserver_entry(owhost, owport)
+                if existing_entry is not None:
+                    raise AbortFlow("already_configured")
+                await self.hass.async_add_executor_job(protocol.proxy, owhost, owport)
+                return self.async_create_entry(title=owhost, data=self.onewire_config)
```

Comment:
Please only wrap the line that can raise in the try... except block. Use an `els:` block if needed.